gliderpy/plotting.py

```diff
@@ -90,3 +90,55 @@ def plot_transect(
     ax.set_ylim(ax.get_ylim()[0], 0)
 
     return fig, ax
+
+
+@register_dataframe_method
+def plot_ctd(
+    df: pd.DataFrame,
+    profile_number: int,
+    var: str,
+    ax: plt.Axes = None,
+    color: str | None = None,
+) -> tuple:
+    """Make a CTD profile plot of pressure vs property
+    depending on what variable was chosen.
+
+    :param profile_number: profile number of CTD
+    :param var: variable to plot against pressure
+    :param ax: existing axis to plot on (default: None)
+    :param color: color for the plot line (default: None)
+    :return: figure, axes
+    """
+    g = df.groupby(["longitude", "latitude"])
+    profile = g.get_group(list(g.groups)[profile_number])
+
+    if ax is None:
+        fig, ax = plt.subplots(figsize=(5, 6))
+        ax.plot(profile[var], profile["pressure"], label=var, color=color)
+        ax.set_ylabel("Pressure")
+        ax.set_xlabel(var)
+        ax.invert_yaxis()
+        ax.legend(loc="lower center", bbox_to_anchor=(0.5, 0.09)
+)
+        return fig, ax
+
+    fig = ax.get_figure()
+
+    # Check if the ax is already a twinx or twiny
+    if hasattr(ax, "twinned_axis"):
+        # If already twinned, get the correct twinned axis
+        ax = ax.twinned_axis
+    else:
+        # Create a new twiny axis and keep reference
+        ax = ax.twiny()
+        ax.twinned_axis = ax
```

Comment:
Do we need this twin ax check and creation?

Reply:
@FloraSauerbronn I'm confused. Do we need this? Why the thumbs down?

Reply:
@ocefpaf 

The problem occurs when calling the legend while using the twiny axis. We have three scenarios:

1. When using just one axis (not referring to a second one like ax2), the legend will only display the last variable plotted if twiny is used, so we need to create twoo box one for each legend and specify where they should be placed.

`
@register_dataframe_method
def plot_ctd(
df: pd.DataFrame,
profile_number: int,
var: str,
ax: plt.Axes = None,
color: str | None = None,
) -> tuple:
"""Make a CTD profile plot of pressure vs property
depending on what variable was chosen.

:param profile_number: profile number of CTD
:param var: variable to plot against pressure
:param ax: existing axis to plot on (default: None)
:param color: color for the plot line (default: None)
:return: figure, axes
"""
g = df.groupby(["longitude", "latitude"])
profile = g.get_group(list(g.groups)[profile_number])

if ax is None:
    fig, ax = plt.subplots(figsize=(5, 6))
    ax.plot(profile[var], profile["pressure"], label=var, color=color)
    ax.set_ylabel("Pressure")
    ax.set_xlabel(var)
    ax.invert_yaxis()
    ax.legend(loc="lower center", bbox_to_anchor=(0.5, 0.07))
    return fig, ax
else:
    ax = ax.twiny()  # Create a new x-axis on top
    ax.plot(profile[var], profile["pressure"], label=var, color=color)
    ax.set_xlabel(var)
    ax.legend(loc="lower center")
    return ax.figure, ax
fig, ax = plot_ctd(df, 0, var="temperature", color="blue")
fig, ax = plot_ctd(df, 0, var="salinity", ax=ax, color="red")`

2. Or calling the legend inside each clouse (if ax is none and else) without defining each ax they would overlap eachother.

3. Or we defined each ax and call the legens like this so they can be in the same box

`@register_dataframe_method
def plot_ctd(
df: pd.DataFrame,
profile_number: int,
var: str,
ax: plt.Axes = None,
color: str | None = None,
) -> tuple:
"""Make a CTD profile plot of pressure vs property
depending on what variable was chosen.

:param profile_number: profile number of CTD
:param var: variable to plot against pressure
:param ax: existing axis to plot on (default: None)
:param color: color for the plot line (default: None)
:return: figure, axes
"""
g = df.groupby(["longitude", "latitude"])
profile = g.get_group(list(g.groups)[profile_number])

if ax is None:
    fig, ax = plt.subplots(figsize=(5, 6))
    lns1 = ax.plot(profile[var], profile["pressure"], label=var, color=color)
    ax.set_ylabel("Pressure")
    ax.set_xlabel(var)
    ax.invert_yaxis()
    
    # Get handles and labels for the legend
    handles, labels = ax.get_legend_handles_labels()
    ax.legend(handles, labels, loc="best")
    return fig, ax
else:
    ax2 = ax.twiny()  # Create a new x-axis on top
    lns2 = ax2.plot(profile[var], profile["pressure"], label=var, color=color)
    ax2.set_xlabel(var)
    
    # Get handles and labels for both axes
    handles1, labels1 = ax.get_legend_handles_labels()
    handles2, labels2 = ax2.get_legend_handles_labels()
    
    # Combine handles and labels
    handles = handles1 + handles2
    labels = labels1 + labels2
    
    # Set the combined legend
    ax.legend(handles, labels, loc="best")
    
    return ax.figure, ax2
Example usage:
fig, ax = plot_ctd(df, 0, var="temperature", color="blue")
fig, ax = plot_ctd(df, 0, var="salinity", ax=ax, color="red")

Reply:
@FloraSauerbronn my question is if we need the twin ax there. We are plotting a single variable. If the user already has one they can:

1. use the same axm
2. provide the twin ax themselves.

The less we try to guess, the less code we have to maintain and less chances for users to shoot themselves in the foot.

Reply:
I see. I thought it was previously suggested that we could plot more than one variable in the same profile plot. But maybe I just imagined that from some other work. Should I remove it?

Reply:
I guess I was not clear. The idea of supporting `ax` as an input is to open for that possibility. However, we don't want to "own" that code. We outsource that to the users b/c they can come up with whatever they want. If we start supporting this we would end up adding so many bells and whistles that: (a) the code would be too complex to maintain and (b) we would never satisfy all users b/c the specific flag they want is not here.

Reply:
@ocefpaf 
Ok, I think I got it now. Just give the user a new axis if needed, and then they can manipulate it as they please. So, only add something like:
`    if ax is None:
        fig, ax = plt.subplots(figsize=(5, 6))
    else:
        fig = ax.get_figure()`

And then proceed with the usual plot features for a simple profile.

Reply:
Yep. Perfect.

Reply:
@ocefpaf 
Done